### PR TITLE
Np 46622 rerun failed cristin objects

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: cristin-import/src/test/java/no/unit/nva/cristin/FakeAmazonSQS.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+import com.amazonaws.services.sqs.model.TagQueueRequest;
+import com.amazonaws.services.sqs.model.TagQueueResult;
+import com.amazonaws.services.sqs.model.UntagQueueRequest;
+import com.amazonaws.services.sqs.model.UntagQueueResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FakeAmazonSQS implements AmazonSQS {
```
Comment (truhacevkir) [inline cristin-import/src/test/java/no/unit/nva/cristin/FakeAmazonSQS.java:62]: Had to copy this over here.

File: template.yaml [hunk 2]
```diff
@@ -472,6 +472,7 @@ Resources:
             Action:
               - s3:DeleteObject
             Resource:
+              - !Sub 'arn:aws:s3:::cristin-import-${AWS::AccountId}/*'
```
Comment (anetteOlli) [inline template.yaml:475]: Supert, at du ikke gikk for stjerne.

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinRerunErrorsEventEmitter.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(CristinRerunErrorsEventEmitter.class);
+    private static final String CRISTIN_IMPORT_BUCKET = ENVIRONMENT.readEnv("CRISTIN_IMPORT_BUCKET");
+    private static final String CRISTIN_ENTRY_QUEUE = ENVIRONMENT.readEnv("CRISTIN_IMPORT_DATA_ENTRY_QUEUE_URL");
+    private final S3Driver s3Driver;
+    private final SqsBatchMessenger batchMessenger;
+
+    @JacocoGenerated
+    public CristinRerunErrorsEventEmitter() {
+        this.s3Driver = new S3Driver(S3Driver.defaultS3Client().build(), CRISTIN_IMPORT_BUCKET);
+        this.batchMessenger = new SqsBatchMessenger(defaultAmazonSQS(), CRISTIN_ENTRY_QUEUE);
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinRerunErrorsEventEmitter.java:46]: use this(new S3Driver(S3Driver.defaultS3Client().build(), CRISTIN_IMPORT_BUCKET), new SqsBatchMessenger(defaultAmazonSQS(), CRISTIN_ENTRY_QUEUE)) instead, that way the default constructor stays in sync with the parameterised constructor.

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinRerunErrorsEventEmitter.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    public CristinRerunErrorsEventEmitter(S3Client s3Client, AmazonSQS sqsClient) {
+        this.s3Driver = new S3Driver(s3Client, CRISTIN_IMPORT_BUCKET);
+        this.batchMessenger = new SqsBatchMessenger(sqsClient, CRISTIN_ENTRY_QUEUE);
+    }
+
+    @Override
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) {
+        var event = getRerunFailedEntriesEvent(inputStream);
+        var errorReports = s3Driver.listAllFiles(event.s3Path());
+        var batchResult = putMessagesOnQueue(errorReports);
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinRerunErrorsEventEmitter.java:58]: Doesn't the batchResult contains information regarding failed events?

Maybe not delete those from s3?

**Review by anetteOlli — CHANGES_REQUESTED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/RerunFailedEntriesEvent.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.net.URI;
+import no.unit.nva.commons.json.JsonSerializable;
+import nva.commons.core.paths.UnixPath;
+import nva.commons.core.paths.UriWrapper;
+
+public record RerunFailedEntriesEvent(URI uri) implements JsonSerializable {
+
+    public UnixPath s3Path() {
+        return UriWrapper.fromUri(uri).getPath().removeRoot();
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/RerunFailedEntriesEvent.java:11]: Ja, har testet, men den returnerer uri med slash foran og liker ikke s3Driver.listFiles funksjonen, så jeg gikk for den varianten her.

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**